src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/view/RecordableFragment.kt

```diff
@@ -0,0 +1,202 @@
+package org.wycliffeassociates.otter.jvm.app.ui.takemanagement.view
+
+import com.github.thomasnield.rxkotlinfx.toObservable
+import com.jfoenix.controls.JFXSnackbar
+import javafx.application.Platform
+import javafx.beans.property.SimpleObjectProperty
+import javafx.event.EventHandler
+import javafx.geometry.Pos
+import javafx.scene.Node
+import javafx.scene.Parent
+import javafx.scene.input.MouseEvent
+import javafx.scene.layout.Pane
+import javafx.scene.layout.VBox
+import org.wycliffeassociates.otter.common.data.workbook.Take
+import org.wycliffeassociates.otter.jvm.app.theme.AppStyles
+import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.TakeContext
+import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.viewmodel.AudioPluginViewModel
+import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.viewmodel.RecordableViewModel
+import org.wycliffeassociates.otter.jvm.app.widgets.progressdialog.progressdialog
+import org.wycliffeassociates.otter.jvm.app.widgets.takecard.TakeCard
+import org.wycliffeassociates.otter.jvm.app.widgets.takecard.events.*
+import org.wycliffeassociates.otter.jvm.utils.onChangeAndDoNow
+import tornadofx.*
+
+abstract class RecordableFragment(protected val recordableViewModel: RecordableViewModel) : Fragment() {
+
+    abstract fun createTakeCard(take: Take): TakeCard
+
+    protected val audioPluginViewModel: AudioPluginViewModel by inject()
+
+    /** Add custom components to this container, rather than root*/
+    protected val mainContainer = VBox()
+
+    protected val lastPlayOrPauseEvent: SimpleObjectProperty<PlayOrPauseEvent?> = SimpleObjectProperty()
+
+    private val draggingNodeProperty = SimpleObjectProperty<Node>()
+
+    // This inner class better organizes the components that need to be added to the derived class
+    /** Use the provided functions to add these components to your derived class */
+    inner class DragComponents {
+        private val dragTargetBottom = VBox().apply {
+            bindVisibleToDraggingNodeProperty()
+        }
+
+        fun dragTargetBottom(runOnNode: (VBox.() -> Unit)? = null): Node = dragTargetBottom.apply {
+            runOnNode?.let { it() }
+        }
```

Comment:
Single expression formats are meant to be a single expression; this is too complex for that style of formatting.

If it was fine, this would apply:
https://kotlinlang.org/docs/reference/functions.html#single-expression-functions

I believe it was referenced in clean code, but basically, how you organize your space is important for readability. We read code top to bottom, and each line left to right. We don't read each line though, often just scan top down across the left hand side, so it's helpful when the leftmost space gives us a lot of context to what the rest of the line does, and we follow further down the line (left to right) if we need more details about that context.

This makes a critical assumption though; that every major new step *begins* on a new line. So in this code example, scanning top down, I see dragTargetBottom.... then the first instruction is "runonNode?..." which is actually incorrect. The fact that the closing } appears to close the function block further adds to misleading a reader just glancing at the code.

```
fun dragTargetBottom(runOnNode: (VBox.() -> Unit)? = null): Node { 
    return dragTargetBottom.apply {
        runOnNode?.let { it() }
    }
}
```

I know this is done elsewhere in the code; I would appreciate if we could all be on the lookout and clean them up as we come across it.

Reply:
Also, @aunger is right, this is really difficult to read; especially with overloading the variable name with the method name.

it *might* help to prefix the private variable with _; so _dragTargetBottom?

But the name is also really unhelpful to begin with; what is it exactly? is it the takeGridDragTarget?

Reply:
I'm going to rework all of this, probably into a separate component, something like DragTakeTarget